File: app/src/test/java/com/mvcoding/financius/data/PageResultTest.java
/*
 * Copyright (C) 2015 Mantas Varnagiris.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 */

package com.mvcoding.financius.data;

import com.mvcoding.financius.BaseTest;
import com.mvcoding.financius.data.paging.Page;
import com.mvcoding.financius.data.paging.PageResult;

import org.junit.Test;

import java.util.Arrays;
import java.util.Collections;

import static org.assertj.core.api.Assertions.assertThat;

public class PageResultTest extends BaseTest {
    @Test public void hasPrevious_returnsFalse_whenPageStartIs0() throws Exception {
        final Page page = new Page(0, 1);
        final PageResult<Object> pageResult = new PageResult<>(page, Collections.emptyList(), false);

        assertThat(pageResult.hasPrevious()).isFalse();
    }

    @Test public void hasPrevious_returnsTrue_whenPageStartMoreThan0() throws Exception {
        final Page page = new Page(1, 1);
        final PageResult<Object> pageResult = new PageResult<>(page, Collections.emptyList(), false);

        assertThat(pageResult.hasPrevious()).isTrue();
    }

    @Test public void hasNext_returnsFalse_whenReturnedItemSizeIsLessThanRequestedSize() throws Exception {
        final Page page = new Page(0, 2);
        final PageResult<Object> pageResult = new PageResult<>(page, Collections.singletonList(new Object()), false);

        assertThat(pageResult.hasNext()).isFalse();
    }

    @Test public void hasNext_returnsTrue_whenReturnedItemSizeEqualsRequestedSize() throws Exception {
        final Page page = new Page(0, 2);
        final PageResult<Object> pageResult = new PageResult<>(page, Arrays.asList(new Object(), new Object()), false);

        assertThat(pageResult.hasNext()).isTrue();
    }
}

File: core-app/src/test/kotlin/com/mvcoding/expensius/paging/PageResultTest.kt
/*
 * Copyright (C) 2015 Mantas Varnagiris.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 */

package com.mvcoding.expensius.paging

import com.natpryce.hamkrest.assertion.assertThat
import com.natpryce.hamkrest.equalTo
import org.junit.Test

class PageResultTest {
    @Test
    fun doesNotHavePreviousPageWhenPageStartsAt0() {
        val pageResult = PageResult<Unit>(Page(0, 5), listOf(), true)

        assertThat(pageResult.hasPreviousPage(), equalTo(false))
    }

    @Test
    fun hasPreviousPageWhenPageStartsAtMoreThan0() {
        val pageResult = PageResult<Unit>(Page(5, 5), listOf(), true)

        assertThat(pageResult.hasPreviousPage(), equalTo(true))
    }

    @Test
    fun doesNotHaveNextPageWhenItemsSizeIsLessThanPageSize() {
        val pageResult = PageResult<Unit>(Page(0, 5), listOf(), true)

        assertThat(pageResult.hasNextPage(), equalTo(false))
    }

    @Test
    fun hasNextPageWhenItemsSizeIsSameAsPageSize() {
        val pageResult = PageResult(Page(0, 2), listOf(Unit, Unit), true)

        assertThat(pageResult.hasNextPage(), equalTo(true))
    }

    @Test
    fun positionIsSameAsPageFirstPosition() {
        val pageResult = PageResult(Page(1, 2), listOf(Unit, Unit), true)

        assertThat(pageResult.position, equalTo(1))
    }
}